apps/shipments/permissions.py

```diff
@@ -65,9 +66,17 @@ def has_permission(self, request, view):
         if nested_shipment:
             # Nested routes do not call has_object_permission for the parent Shipment,
             # so we must check object permissions here
-            shipment = Shipment.objects.get(id=view.kwargs.get('shipment_pk'))
+            shipment = Shipment.objects.get(id=nested_shipment)
             return self.has_object_permission(request, view, shipment)
 
+        if nested_device:
+            # Nested device routes need to be ensured that they have a shipment associated with the device
+            if not settings.PROFILES_ENABLED:
```

Comment:
I don't think this PROFILES_ENABLED check belongs here. You can add a conditional permissionclass declaration on the view, which is how we usually handle PROFILES_ENABLED branching.